src/aggregate/aggregate.h

```diff
@@ -66,7 +66,10 @@ typedef enum {
 #define IsCount(r) ((r)->reqflags & QEXEC_F_NOROWS)
 #define IsSearch(r) ((r)->reqflags & QEXEC_F_IS_SEARCH)
 #define IsProfile(r) ((r)->reqflags & QEXEC_F_PROFILE)
-#define RunInThread(r) (RSGlobalConfig.threadsEnabled && RSGlobalConfig.numWorkerThreads && IsSearch(r))
+// These macro should be used only by the main thread since configuration can be changed while running in 
+// backgroud.
+#define RUN_IN_THREAD (RSGlobalConfig.threadsEnabled && RSGlobalConfig.numWorkerThreads)
+#define RunSearchInThread(r) (RUN_IN_THREAD && IsSearch(r))
```

Comment:
I think we should keep this macro generic and in the future, we can edit its condition when we need and not modify the flows that use it (for example when we support aggregate in the BG we can just edit the condition and not everyone who want to check if it should run in BG or not)